### DISCUSSION TITLE
Speed of querying data

### DISCUSSION THREAD

**Discussion by balexus** [questdb/questdb#1973] [Show and tell]
I want to get 2.000.000 records from table via simple SQL request like 
**select * from table**

Query took several milliseconds and it is OK but  when i use REST API /exec command for it i get result in 3 or 4 seconds. So getting result takes 3 or 4 seconds. 

**questdb and client are located on same computer**

Is there any way to get 2.000.000 records in 0.1 seconds ?

**Comment by glasstiger** [comment]
Hi balexus, HTTP adds a lot of overhead on top of the query.
If you need that kind of latency I suggest you try to use QuestDB embedded in your application.
Could you provide a bit more detail on your use case and architecture, please?

**Reply by balexus** [reply]
Thanks glasstiger.

I use questdb 6.2 
CentOS, 
PHP 7.2

I use REST API from PHP via sockets

**Reply by ideoma** [reply]
Hey, you can measure pure QuestDB timing of REST call using 

```
curl -G --data-urlencode "query=SELECT * FROM table"  http://localhost:9000/exec >> /dev/null
```

I would be expected to be something around 200Mb / sec and limit is mostly with CPU serializing and sending the data. If your data serialized size is 20Mb then it's possible to get it in 0.1 sec using REST.

**Reply by balexus** [reply]
i have measured time for pure QuestDB timing of REST call using curl as above
it takes 1 - 1.3 seconds 
result size is 75MB
Can time be quicker or it is limit ?

**Reply by ideoma** [reply]
It can be for sure as I said it can be 200Mb/sec on a beefy hardware.
